packages/bot/src/services/parseSetupMessage.ts

```diff
@@ -2,10 +2,17 @@ import { Message } from 'discord.js'
 
 // parse received setup message
 export function parseSetupMessage (message: Message) {
-  const chunks = message.content.replace('  ', ' ').split(' ')
+  // const chunks = message.content.replace('  ', ' ').split(' ')
   const guildId = message.guild?.id
-  const daoName = chunks[1]
-  const roleAllowed = chunks.slice(2, chunks.length).join(' ')
+  // Split between the `!setup` part, the DAO name, and the allowed role
+  const chunks = message.content.replace('  ', ' ').split(/\@|!setup/)
+    ? message.content.replace('  ', ' ').split(/\@|!setup/)
+    : message.content.replace('  ', ' ').split(/\<@|!setup/)
+  // The allowed role is always the last chunk, so we can simply pop it
+  const roleAllowed = '@' + chunks.pop()?.trim()
```

Comment:
If there's a chance that roles come in the `<@whatever>` format, we may need to get rid of the `>` sign as well:
```suggestion
  const roleAllowed = '@' + chunks.pop()?.replace('>', '').trim()
```

Reply:
We need to include that sign